### ISSUE TITLE
SCAlertView only showing once after segue ?

### ISSUE THREAD

**Issue by OmarAlKhamis** [vikmeup/SCLAlertView-Swift#219]
Hello guys , 

I really , really like the design of those awesome alertViews.
My issue is as follows:

My code consist of two viewControllers which are calling each other back and forth,
MainScreenViewController
GamePlayViewController

MainScreenViewController calls performSegueWithIdentifier using a button with @IBAction to GamePlayViewController

in GamePlayViewController.viewDidLoad() i did the following :

`
...
        let alert = SCLAlertView(appearance: SCLAlertView.SCLAppearance(showCloseButton: false))
        alert.showWait(Strings.MainScreenViewController.JustWaitTitle, subTitle: Strings.MainScreenViewController.JustWaintSubTitle, duration: Constants.Timer.InitialCountingDown)
...
`

basically an alert with 5 seconds count down , then it closes, great.

Now , after extra 10 seconds , with NSTimer , the GamePlayViewController calls performSegueWithIdentifier to the MainScreenViewController 

**Now the second time**  , when i press the start button on the MainScreenViewController , the segue happens, but no alert is showing at all.

I tried making the alert code in the button action before the performSegue instead but same problem ...

Please help :) , i really wanna use those great alerts

_Ps.:_

I tried disabling the segues and trying with one viewController only and it works , the alert shows whenever i call .showWait(...) , 
That's why i thing the problem has to do with segues and changing the viewController

**Comment by OmarAlKhamis** [comment]
**_Solved !_**

I just had to call it in viewDidAppear instead of viewDidLoad ...
can anybody explain why ?

**Comment by TheOriginalTrubador** [comment]
Yeah- viewDidLod is only called on initial load- viewDidAppear is called
each time the controller appears. First time you hit it is onLoad, second
isn't old since the controller is loaded in the hierarchy it's you've
segued into another.

Regards

Ian Bray

Sent from my iPhone - Apologies for any typos

On Jul 27, 2016, at 21:24, OmarAlKhamis <notifications@github.com> wrote:

*Solved !*

I just had to call it in viewDidAppear instead of viewDidLoad ...
can anybody explain why ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/vikmeup/SCLAlertView-Swift/issues/219#issuecomment-235770887>,
or mute the thread
